posthog/migrations/0862_alter_team_session_recording_retention_period.py

```diff
@@ -0,0 +1,50 @@
+from django.db import migrations, models
+
+import structlog
+
+logger = structlog.get_logger(__name__)
+
+
+def migrate_replay_retention_period(apps, schema_editor):
+    Team = apps.get_model("posthog", "Team")
+
+    teams_to_migrate = []
+
+    for team in (
+        Team.objects.filter(session_recording_retention_period="legacy")
+        .only("id", "session_recording_retention_period")
+        .iterator(chunk_size=100)
+    ):
+        try:
+            team.session_recording_retention_period = "30d"
+            teams_to_migrate.append(team)
+        except Exception as e:
+            # If anything fails for a team, skip it and continue with others
+            logger.error("replay_retention_period_migration_failed", team_id=team.id, error=str(e), exc_info=True)
+            continue
+
+    Team.objects.bulk_update(
```

Comment:
🤔 if i read this right then it's one big bulk update at the end, not one per 100 in the iterator
so we could get stuck in a migration loop where we can't complete, then we retry but nothing changed, and goto 10

would be better imho to manually bulk update every X teams and that way even if there are e.g. OOM fails we'd still make progress... am sure there's an example...